docker/indexer/stages/preparation/preparation_test.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2022 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+// Package preparation provides functionality to extract tags, branches and commits from repository configurations.
+package preparation
+
+import "testing"
+
+func Test_tagToStandardVersion(t *testing.T) {
+	type args struct {
+		tag string
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "",
+			args: args{
+				tag: "cares-1_5_0",
+			},
+			want: "1.5.0",
+		},
+		{
+			name: "",
+			args: args{
+				tag: "cares-1_5_0_alpha",
+			},
+			want: "1.5.0.alpha",
+		},
+		{
+			name: "",
+			args: args{
+				tag: "cares-1_5_0-alpha4",
+			},
+			want: "1.5.0.alpha4",
+		},
+		{
+			name: "",
+			args: args{
+				tag: "cares-1_5_0-6alpha4",
+			},
+			want: "1.5.0.6.4",
```

Comment:
Should this be 1.5.0.6.alpha4 ?

Reply:
The python version will return `1.5.0.6.alpha4`, but I'm not sure how to replicate it in golang without lookbehinds, at least without changing behavior somewhere else, or complicating it by doing multiple passes with regexes.

Reply:
We can probably just completely remove that negative lookbehind. I think that was just to capture a very rare edge case with things like "arc" getting interpreted as a version component. 

Also more broadly, we should just compute this at the API serving time rather than record it here. Then we can refine/fix this whenever without having to re-index. That can come later if it's complicated to do here.

Reply:
(And doing this at the API serving layer also fixes the Python/Go discrepancy)

Reply:
Removed this entirely to favor parsing and filtering it at API serving time. PTAL